modules/m_stats.c

```diff
@@ -1619,19 +1621,27 @@ stats_l_client(struct Client *source_p, struct Client *target_p,
 
 	else
 	{
+		/* fire the doing_stats_show_idle hook to allow modules to tell us whether to show the idle time */
+		hook_data_client_approval hdata_showidle;
+
+		hdata_showidle.client = source_p;
+		hdata_showidle.target = target_p;
+		hdata_showidle.approved = 1;
+
+		call_hook(doing_stats_show_idle_hook, &hdata_showidle);
 		sendto_one_numeric(source_p, RPL_STATSLINKINFO, Lformat,
 				   show_ip(source_p, target_p) ?
 				    (IsUpper(statchar) ?
 				     get_client_name(target_p, SHOW_IP) :
 				     get_client_name(target_p, HIDE_IP)) :
 				    get_client_name(target_p, MASK_IP),
-				    (int) rb_linebuf_len(&target_p->localClient->buf_sendq),
-				    (int) target_p->localClient->sendM,
-				    (int) target_p->localClient->sendK,
-				    (int) target_p->localClient->receiveM,
-				    (int) target_p->localClient->receiveK,
+				    hdata_showidle.approved ? (int) rb_linebuf_len(&target_p->localClient->buf_sendq): 0,
+				    hdata_showidle.approved ? (int) target_p->localClient->sendM : 0,
```

Comment:
`0` is a valid value. I'd prefer to see a distinct value like `*` or `-1` here when it was hidden from view. This command is already restricted in who can run it. We shouldn't make the output *wrong* for operators without the privilege.

Reply:
Hiding `317` entirely was the original behavior but then was changed per https://github.com/solanum-ircd/solanum/pull/220#discussion_r659179325 and https://github.com/solanum-ircd/solanum/pull/220#issuecomment-879386882. We eventually ended up putting the `is using a private idle time` bit into the `WHOIS` response to make that less confusing. As far as `stats` is concerned, I think it does make sense to use `-1` here.